src/dataverk/datapackage.py

```diff
@@ -91,102 +97,41 @@ def path(self):
     def url(self):
         return self._datapackage_metadata.get("url")
 
-    def _get_schema(self, df, path, resource_name, resource_description, format, compress, dsv_separator, spec):
-        fields = []
-
-        for name, dtype in zip(df.columns, df.dtypes):
-            if str(dtype) == 'object':
-                dtype = 'string'
-            else:
-                dtype = 'number'
-
-            description = ''
-            if spec and spec.get('fields'):
-                spec_fields = spec['fields']
-                for field in spec_fields:
-                    if field['name'] == name:
-                        description = field['description']
-
-            fields.append({'name': name, 'description': description, 'type': dtype})
-
-        mediatype = self._media_type(format)
-
-        return {
-            'name': resource_name,
-            'description': resource_description,
-            'path': self._resource_path(path, resource_name, format, compress),
-            'format': format,
-            'dsv_separator': dsv_separator,
-            'compressed': compress,
-            'mediatype': mediatype,
-            'schema': {'fields': fields},
-            'spec': spec
-        }
+    def add_resource(self, resource: Any, resource_type: str, resource_name: str = "",
+                     resource_description: str = "", spec: dict = None):
```

Comment:
Kanskje defaulte resource_type til "ResourceType.DF.value"? Selv om vi åpner for flere typer så vil det fremdeles oftest være en pandas dataframe vi legger til. Og, dersom vi ikke defaulter til "ResourceType.DF.value" vil alle notebooks og applikasjoner som bruker dataverk knekke da ingen av disse setter resource_type per nå.

Reply:
Godt poeng!!! Fikser nå